### PR TITLE
fix(pymongo): update pymongo integration for compatibility with 4.5.x

### PR DESCRIPTION
**Pull request by emmettbutler**
Pymongo 4.5.0 [changed the name](https://github.com/mongodb/mongo-python-driver/pull/1329) of the `get_socket` function on which our integration is based. This pull request adapts the integration to work with version 4.5.0 by changing the expected function name based on the version.

Fixes https://github.com/DataDog/dd-trace-py/issues/6723

## Checklist

- [x] Change(s) are motivated and described in the PR description.
- [x] Testing strategy is described if automated tests are not included in the PR.
- [x] Risk is outlined (performance impact, potential for breakage, maintainability, etc).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] [Library release note guidelines](https://ddtrace.readthedocs.io/en/stable/releasenotes.html) are followed. If no release note is required, add label `changelog/no-changelog`.
- [x] Documentation is included (in-code, generated user docs, [public corp docs](https://github.com/DataDog/documentation/)).
- [x] Backport labels are set (if [applicable](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting))

## Reviewer Checklist

- [x] Title is accurate.
- [x] No unnecessary changes are introduced.
- [x] Description motivates each change.
- [x] Avoids breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [x] Testing strategy adequately addresses listed risk(s).
- [x] Change is maintainable (easy to change, telemetry, documentation).
- [x] Release note makes sense to a user of the library.
- [x] Reviewer has explicitly acknowledged and discussed the performance implications of this PR as reported in the benchmarks PR comment.
- [x] Backport labels are set in a manner that is consistent with the [release branch maintenance policy](https://ddtrace.readthedocs.io/en/latest/contributing.html#backporting)

### CODE REVIEW

File: ddtrace/contrib/pymongo/patch.py [hunk 1]
```diff
@@ -28,6 +28,9 @@
 # Original Client class
 _MongoClient = pymongo.MongoClient
 
+VERSION = pymongo.version_tuple
+CHECKOUT_FN_NAME = "get_socket" if VERSION < (4, 5) else "checkout"
+
```
Comment (majorgreys) [inline ddtrace/contrib/pymongo/patch.py:33]: Make these private

File: riotfile.py [hunk 2]
```diff
@@ -670,13 +670,13 @@ def select_pys(min_version=MIN_PYTHON_VERSION, max_version=MAX_PYTHON_VERSION):
...
                 Venv(
-                    pys=select_pys(min_version="3.7", max_version="3.9"), pkgs={"pymongo": ["~=3.11", "~=4.0", latest]}
+                    pys=select_pys(min_version="3.7", max_version="3.9"), pkgs={"pymongo": ["~=3.11", "~=4.5", latest]}
                 ),
                 Venv(
                     # pymongo added support for Python 3.10 in 3.12.1
                     # pymongo added support for Python 3.11 in 3.12.3
                     pys=select_pys(min_version="3.10"),
-                    pkgs={"pymongo": ["~=3.12.3", "~=4.0", latest]},
+                    pkgs={"pymongo": ["~=3.12.3", "~=4.5", latest]},
```
Comment (majorgreys) [inline riotfile.py:679]: Why was this bump necessary? Did the patching code work fine with `pymongo>=4.0,<4.5`? Shouldn't `~=4.0` resolve to the same version as `~=4.5`?

Comment (emmettbutler) [reply riotfile.py]: Yeah, it does. Not sure why it wasn't working yesterday.

**Review by tabgok — APPROVED** [review]

**Review by majorgreys — APPROVED** [review]

**Review by brettlangdon — DISMISSED** [review]

File: releasenotes/notes/pymongo-get-socket-808421288343ab81.yaml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+---
+upgrade:
```
Comment (brettlangdon) [inline releasenotes/notes/pymongo-get-socket-808421288343ab81.yaml:2]: fix? 🤔

**Review by tabgok — APPROVED** [review]

File: releasenotes/notes/pymongo-get-socket-808421288343ab81.yaml (hunk 3 above)
Comment (tabgok) [reply releasenotes/notes/pymongo-get-socket-808421288343ab81.yaml:2]: agreed

**Review by tabgok — APPROVED** [review]

**Review by brettlangdon — DISMISSED** [review]

**Review by tabgok — APPROVED** [review]

**Review by brettlangdon — APPROVED** [review]

**Pull request merged**